c7n/resources/iam.py

```diff
@@ -376,14 +376,22 @@ class IamRoleInlinePolicy(Filter):
     permissions = ('iam:ListRolePolicies',)
 
     def _inline_policies(self, client, resource):
-        return len(client.list_role_policies(
-            RoleName=resource['RoleName'])['PolicyNames'])
+        policies = client.list_role_policies(
+            RoleName=resource['RoleName'])['PolicyNames']
+        resource['InlinePolicies'] = policies
```

Comment:
when we're adding additional attributes to a resource via custodian, general rule is to always prefix those, so they can be disambiguated from native attributes (either present or those that may be appear in future). we've unfortunately been a little inconsistent on the prefix `c7n:` and `c7n-` but both need escaping for jmespath expressions, preference at the moment is `c7n:`

Reply:
sounds good, I was thinking that this was kind of doing an augment on the resource so I was dropping the c7n: prefix, i'll s/InlinePolicies/c7n:InlinePolicies shortly

Reply:
yeah, the distinction is something we evolved to. the only time we have exceptions on the prefix are s3 augments (which predate), and tags for resources that lack tags.